### DISCUSSION TITLE
Interested in porting JSX-Control-Statements plugin to SWC Compiler

### DISCUSSION THREAD

**Discussion by chrisrichard** [vercel/next.js#33344] [Help]
https://github.com/AlexGilleran/jsx-control-statements

Are there any examples of similar transforms already implemented or background info on where to add this kind of thing?

**Comment by alienfacepalm** [comment]
We love this plugin so much, it's preventing us from moving to SWC. React really should just have conditional JSX, it's wack that it doesn't. I agree with this.

**Comment by chrisrichard** [comment]
FYI https://github.com/intpp/swc-plugin-jsx-control-statements

Haven't tried it yet but looks promising.

**Reply by alienfacepalm** [reply]
Nice. Been digging Solid lately because it's just built in. https://docs.solidjs.com/guides/tutorials/getting-started-with-solid/control-flow
